pkg/services/dashboards/database/database.go

```diff
@@ -574,7 +574,7 @@ func GetAlertsByDashboardId2(dashboardId int64, sess *sqlstore.DBSession) ([]*mo
 }
 
 func (d *DashboardStore) updateAlerts(ctx context.Context, existingAlerts []*models.Alert, alerts []*models.Alert, log log.Logger) error {
-	return d.sqlStore.WithTransactionalDbSession(ctx, func(sess *sqlstore.DBSession) error {
+	return d.sqlStore.WithDbSession(ctx, func(sess *sqlstore.DBSession) error {
```

Comment:
Does changing this make any difference? Since the existing session is retrieved either way, the only difference is an extra boolean check on the transactionOpen variable, right?

Reply:
I think that it will start [a nested transaction](https://github.com/grafana/grafana/blob/bd50fd1606c1b5ca8fdf273902455834d3e5447b/pkg/services/sqlstore/session.go#L51)

Reply:
We had WithTx(WithTx()), so the inner one would not start a new transaction because it would [return at line 46](https://github.com/grafana/grafana/blob/bd50fd1606c1b5ca8fdf273902455834d3e5447b/pkg/services/sqlstore/session.go#L46).

Either way, removing the call is the right thing to do.